syntax = "proto3";
package moonalt;

message RawFrame {
  uint64 frame_id     = 1;
  uint32 width        = 2;
  uint32 height       = 3;
  string pixel_format = 4;  // "BGR24"
  bytes  data         = 5;  // H*W*3
}

message MaskFrame {
  uint64 frame_id     = 1;
  uint32 width        = 2;
  uint32 height       = 3;
  string pixel_format = 4;  // "GRAY8"
  bytes  data         = 5;  // H*W
}

message BoundingBox {
  float x1 = 1;
  float y1 = 2;
  float x2 = 3;
  float y2 = 4;
}

message Boxes {
  uint64 frame_id        = 1;
  repeated BoundingBox b = 2;
}

message YoloPacket {
  string schema = 1;   // "moonalt.yolo_packet_v1"
  RawFrame raw  = 2;
  MaskFrame mask = 3;
  Boxes boxes   = 4;
}
